### PR TITLE
refactor: build chain to rollup

### LINKED ISSUE #102: [Tech Debt] Update and simplify build chain for all packages
## Context

- Part of: https://www.notion.so/walletconnect/Tax-Day-2023-rather-Tax-Week-59906457d45a4214960792a1c3cf1b25?pvs=4
- Branch: https://github.com/WalletConnect/walletconnect-utils/tree/refactor/build-chain

## Steps

- [x] Update and standardise the build chain to use rollup in the same manner as in the monorepo: https://github.com/WalletConnect/walletconnect-monorepo/blob/v2.0/rollup.config.js. This provides us with a standardised build chain across the board that can be changed in the same way in the future.
- [x] Validate that new package builds can be consumed downstream as before
- [x] Drop `tslib` as a dependency across packages
- [x] Remove unused husky hooks
- [x] Fix `lint` step in all pkgs
- [x] Fix `build` step in all pkgs
- [x] Fix `test` step in all pkgs
- [x] Make prettier step explicit in scripts and CI
- [x] CI: parallelize `lint + build + test` job to  rather than running as single job
- [x] Hoist and fix up eslint configuration at monorepo level

### PR DESCRIPTION
**Pull request by bkrem**
## Context

- Resolves #102 

## Summary of Changes

- Applied the same rollup-based build chain we use in the monorepo to all of these util pkgs, which means:
	- Less cognitive load when switching between the main `walletconnect-monorepo` and this satellite utils monorepo
	- Changes/Fixes/Updates to the build chain in one repo can be applied to the other
- Removed cruft (unused husky hooks, inlined prettier configs)
- Removed `tslib` from pkg `dependencies` which the prior build chain was reliant on in its outputs
- Standardised & deduped ESLint and Prettier devDeps and configs at the root level
- Split the monolithic CI workflow into multiple jobs

## How has this been tested?

- Did some dogfooding with multiple packages consumed by `@walletconnect/core`
- Needs another run of canary testing before merge

### CODE REVIEW

File: misc/encoding/src/index.ts [hunk 1]
```diff
@@ -155,6 +153,7 @@ export function binaryToNumber(bin: string): number {
 // -- Validators ----------------------------------------- //
 
 export function isBinaryString(str: any): boolean {
+  // eslint-disable-next-line prefer-regex-literals
```
Comment (devceline) [inline misc/encoding/src/index.ts:156]: Wouldn't we not need this ignore anymore with the new lint rules

File: package-lock.json
Comment (devceline) [inline package-lock.json:1]: Running `npm i` on the root of the directory (or within subpackages) causes the `package-lock` to get updated (Ran with node 16 and 18)

Comment (bkrem) [reply package-lock.json:1]: Right but it resolves itself if you bootstrap the monorepo via lerna like so:

```bash
# install monorepo/root deps
npm i

# bootstrap all the packages
npm run bootstrap
```

Also noted in the readme instructions but will see what can be done to make this easier to work with.

File: misc/encoding/src/index.ts (hunk 1 above)
Comment (bkrem) [reply misc/encoding/src/index.ts]: Good catch thank you 👍

**Review by devceline — APPROVED** [review]
LGTM

**Review by ganchoradkov — APPROVED** [review]
Awesome, much-awaited update 💯 🥇

**Pull request merged**